### PR TITLE
Refactor the graph generation (from edge list) code

### PR DESCRIPTION
**Pull request by seunghwak**
- Move the graph generation function from ```tests/utilities``` to ```src/generator```.
- Use std::optional to take the optional vertex list parameter instead of using two separate functions.

### CODE REVIEW

**Comment by seunghwak** [comment]
@ChuckHastings This is relevant to your test infrastructure update work and you may want to take a look.

**Review by ChuckHastings — APPROVED** [review]
std::optional sure makes the code more compact.

**Comment by rlratzel** [comment]
@seunghwak Thank you for renaming `generate_graph_from_edgelist` to `create_graph_from_edgelist` (and moving it to `structure` too).  I'll review it more thoroughly tomorrow but it looks good to me so far.

File: cpp/include/experimental/graph_functions.hpp [hunk 1]
```diff
@@ -68,6 +74,7 @@ template <typename vertex_t, typename edge_t, bool multi_gpu>
 std::enable_if_t<multi_gpu,
                  std::tuple<rmm::device_uvector<vertex_t>, partition_t<vertex_t>, vertex_t, edge_t>>
 renumber_edgelist(raft::handle_t const& handle,
+                  std::optional<std::tuple<vertex_t const*, vertex_t>> optional_local_vertex_span,
```
[the remark this answers is not in the document]
Comment (seunghwak) [reply cpp/include/experimental/graph_functions.hpp:77]: So, are we planning to directly expose this to cython?

https://github.com/rapidsai/cugraph/blob/7cb858345bbf020c862b5999de8c8081cd880d56/cpp/src/utilities/cython.cu#L1079
AFAIK, we have ```call_renumber``` function, and this function is exposed to cython.

File: cpp/include/experimental/graph_functions.hpp [hunk 2]
```diff
@@ -393,5 +318,48 @@ extract_induced_subgraphs(
...
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          bool store_transposed,
+          bool multi_gpu>
+std::tuple<cugraph::experimental::graph_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu>,
+           rmm::device_uvector<vertex_t>>
+create_graph_from_edgelist(
+  raft::handle_t const& handle,
+  std::optional<std::tuple<vertex_t const*, vertex_t>> optional_vertex_span,
```
[the remark this answers is not in the document]
Comment (aschaffer) [reply cpp/include/experimental/graph_functions.hpp:357]: Okay, just making sure we account for that. Feel free to resolve this topic. Thank you.

**Review by aschaffer — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**